src/FSharp.Core/array.fsi

```diff
@@ -3252,6 +3252,301 @@ module Array =
+        /// <param name="array">The input array.</param>
+        ///
+        /// <exception cref="T:System.ArgumentNullException">Thrown when the input array is null.</exception>
+        /// <exception cref="T:System.ArgumentException">Thrown when the input array is empty.</exception>
+        ///
+        /// <returns>The final result of the reductions.</returns>
+        ///
+        /// <example id="para-reduce-1">
+        /// <code lang="fsharp">
+        /// let inputs = [| 1; 3; 4; 2 |]
+        ///
+        /// inputs |> Array.Parallel.reduce (fun a b -> a + b)
+        /// </code>
+        /// Evaluates to <c>1 + 3 + 4 + 2</c>. However, the system could have decided to compute (1+3) and (4+2) first, and then put them together.
+        /// </example>
+
+        [<CompiledName("Reduce")>]
+        [<Experimental("Experimental library feature, requires '--langversion:preview'")>]
+        val inline reduce: reduction: ('T -> 'T -> 'T) -> array: 'T[] -> 'T
+
+        /// <summary>Applies a projection function to each element of the array in parallel, reducing elements in each thread with a dedicated 'reduction' function.
+        /// After processing entire input, results from all threads are reduced together.
+        /// Raises ArgumentException if the array is empty.</summary>
+        /// <remarks>The order of processing is not guaranteed. For that reason, the 'reduction' function argument should be commutative.
+        /// (That is, changing the order of execution must not affect the result) </remarks>
+        ///
+        /// <param name="projection">The function to project from elements of the input array</param>
+        /// <param name="reduction">The function to reduce a pair of projected elements to a single element.</param>
+        /// <param name="array">The input array.</param>
+        ///
+        /// <exception cref="T:System.ArgumentNullException">Thrown when the input array is null.</exception>
+        /// <exception cref="T:System.ArgumentException">Thrown when the input array is empty.</exception>
+        ///
+        /// <returns>The final result of the reductions.</returns>
+        ///
+        /// <example id="para-reduceBy-1">
+        /// <code lang="fsharp">
+        /// let inputs = [| "1"; "3"; "4"; "2" |]
+        ///
+        /// inputs |> Array.Parallel.reduceBy  (fun x -> int x) (+)
+        /// </code>
+        /// Evaluates to <c>1 + 3 + 4 + 2</c>. However, the system could have decided to compute (1+3) and (4+2) first, and then put them together.
+        /// </example>
+
+        [<CompiledName("ReduceBy")>]
+        [<Experimental("Experimental library feature, requires '--langversion:preview'")>]
+        val reduceBy: projection: ('T -> 'U) -> reduction: ('U -> 'U -> 'U) -> array: 'T[] -> 'U
+
+        /// <summary>Returns the greatest of all elements of the array, compared via Operators.max on the function result.</summary>
```

Comment:
I guess the summary should be at least a bit different from the `max` function.

Reply:
What would "a bit" mean?
The fact that it is Parallel is implied by the module that is being called.

Everything else is technically correct the same way as it is in Array.max.


I have never used it, by XML doc comments allow some kind of reuse via reference to avoid repeating snippets.

Reply:
Yes, max vs maxBy should have different summary.